docs/user-guide/migrating-to-4.0.0.md

```diff
@@ -0,0 +1,187 @@
+}
+```
+
+## <a name="config-validation"/> Unrecognized properties in config files will cause a fatal error
+
+When creating a config, users sometimes make typos or misunderstand how the config is supposed to be structured. Previously, ESLint did not validate the properties of a config file, so a typo in a config could be very tedious to debug. Starting in 4.0.0, ESLint will raise an error if a property in a config file is unrecognized or has the wrong type.
+
+**To address:** If you see a config validation error after upgrading, verify that your config doesn't contain any typos. If you are using an unrecognized property, you should be able to remove it from your config to restore the previous behavior.
+
+## <a name="file-not-found-errors"/> Referencing a missing file with the CLI will cause a fatal error
+
+Previously, ESLint would raise no errors when invoked with missing files:
+
+```bash
+eslint 'missing-file.js' 'missing-folder/' 'empty-folder/**/*.js'
+```
+
+This caused confusion for users, because it was possible to set up a broken linting task that would always succeed if there was a typo in the file/folder name and no files were matched. Starting in 4.0, referencing a file that does not exist or a glob that matches no files from the CLI is now a fatal error.
+
+**To address:** If you see an error about missing files after upgrading, verify that all of the files that you're linting actually exist. Remove any files that don't exist from the command-line arguments.
+
+## <a name="eslintignore-patterns"/> .eslintignore patterns are now resolved from the location of the file
+
+Due to a bug, glob patterns in an `.eslintignore` file were previously resolved from the current working directory of the process, rather than the location of the `.eslintignore` file. Starting in 4.0, patterns in an `.eslintignore` file will be resolved from the `.eslintignore` file's location.
+
+**To address:** If you use an `.eslintignore` file and you frequently run eslint from somewhere other than the project root, it's possible that the patterns will be matched differently. You should update the patterns in the `.eslintignore` file to ensure they are relative to the file, not to the working directory.
+
+## <a name="padded-blocks-defaults"/> The `padded-blocks` rule is more strict by default
+
+By default, the [`padded-blocks`](/docs/rules/padded-blocks) rule will now enforce padding in class bodies and switch statements. Previously, the rule would ignore these cases unless the user opted into enforcing them.
+
+**To address:** If this change results in more linting errors in your codebase, you should fix them or reconfigure the rule.
+
+## <a name="space-before-function-paren-defaults"/> The `space-before-function-paren` rule is more strict by default
+
+By default, the [`padded-blocks`](/docs/rules/padded-blocks) rule will now enforce spacing for async arrow functions. Previously, the rule would ignore these cases unless the user opted into enforcing them.
```

Comment:
typo here: you are referencing the `padded-blocks` rule instead of the `space-before-function-paren` rule